ui/pages/confirmations/components/confirm/info/shared/transaction-details/transaction-details.tsx

```diff
@@ -65,18 +65,22 @@ export const OriginRow = () => {
 export const RecipientRow = ({ recipient }: { recipient?: Hex } = {}) => {
   const t = useI18nContext();
   const { currentConfirmation } = useConfirmContext<TransactionMeta>();
-  const to = recipient ?? currentConfirmation?.txParams?.to;
-  const isBatch = currentConfirmation?.type === TransactionType.batch;
+  const { nestedTransactions, txParams, chainId, id } =
+    currentConfirmation ?? {};
+  const { from, to: txTo } = txParams ?? {};
+  const to = recipient ?? txTo;
+
+  const isBatch =
+    isBatchTransaction(nestedTransactions) &&
+    to?.toLowerCase() === from.toLowerCase();
 
   if (!to || !isValidAddress(to)) {
     return null;
   }
 
-  const { chainId } = currentConfirmation;
-
   return (
     <ConfirmInfoAlertRow
-      ownerId={currentConfirmation.id}
+      ownerId={id}
```

Comment:
Minor, is it worth a quick unit test to check the logo is only rendered if from equals to?

Reply:
Good point, added some coverage for this part.